overlord/hookstate/hooks.go

```diff
@@ -246,11 +246,11 @@ func (h *gateAutoRefreshHookHandler) Done() (err error) {
 			// we have HintInhibitedGateRefresh lock already when running the hook, change
 			// it to HintInhibitedForRefresh.
 			// Also let's reuse inhibit info that was saved in Before().
-			_, inhibitInfo, err := runinhibit.IsLocked(snapName)
+			_, inhibitInfo, err := runinhibit.IsLocked(snapName, st.Unlocker())
 			if err != nil {
 				return err
 			}
-			if err := runinhibit.LockWithHint(snapName, runinhibit.HintInhibitedForRefresh, inhibitInfo); err != nil {
+			if err := runinhibit.LockWithHint(snapName, runinhibit.HintInhibitedForRefresh, inhibitInfo, st.Unlocker()); err != nil {
```

Comment:
this is going to unlock/lock the state twice quickly, it feels like it would be better later to add a single operation that does this

Reply:
updated, thanks for catching this